### PR TITLE
chore: close TASK-17962/17964 follow-up batches — seam tests, provenance, pluralization, documented rulings

### PR DESCRIPTION
**Pull request by rmusser01**
## Summary

Closes both follow-up batches filed during the workspace-modal (#1809) and project-skills (#1810) arcs, after reconciling their ACs against everything later PRs already landed.

**Reconciled as already-covered (with evidence):** the Library activation-failure seam test (shipped in #1809's final fix wave) and the stale-`_folder_discoveries` cleanup (shipped in #1810's fix wave, pinned by test).

**Newly implemented (TASK-17962):** Console + Settings activation-failure seam tests (real registry, surface-specific assertions); a real-keypress Enter fast-path test; a real-TOCTOU `failed_folders` test (`rmtree` between Add and Create, exercising the retry machinery); checkbox-state-survives-recompose via real click; `WorkspaceCreateResult | None` typing on all three surface callbacks; and restored per-surface `description` provenance ("Created from Console/Settings/Library") via a modal parameter — RED-proven.

**Newly implemented (TASK-17964):** offer-modal footer tests (`Skipped:` / "N more not shown"); a checkbox-escape assertion proving `[red]evil[/red]` renders literally (the old shared fixture's unquoted markup silently degraded to an empty description via YAML — new properly-quoted fixture); proper pluralization of the folder-row annotation.

**Documented rulings (not implemented, rationale in task notes + code comments):** off-thread discovery is declined because the synchronous folder validator stats the same filesystem *before* discovery — a dead mount hangs there first, so async discovery alone cannot help; the hung-import dismissal trade-off is accepted because the importer is local-filesystem-only and the inert-while-in-flight guards exist to prevent silent partial-import discard.

Also recorded: a generalizable Textual trap — overriding `on_mount` in a test-harness subclass does NOT shadow the base handler (message dispatch walks the full MRO and calls both), verified against Textual 8.2.8 internals.

## Verification

CI intentionally cancelled — verified locally: `Tests/Workspaces/` 279 passed, `Tests/Skills/` 436 passed, named UI files 29 passed (+1 pre-existing failure — `test_single_item_handoff_gates_on_the_selected_row_not_the_aggregate` — reproduced byte-for-byte on unmodified origin/dev in a throwaway worktree; untouched file, flagged for whoever owns dev reds), 52,305-test collect sweep clean. RED-first tees for the behavior changes.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
## Summary by cubic
Restores per-surface provenance for workspace creation and fixes folder-row pluralization; documents rulings on discovery/import dismissal and adds seam tests that pin activation failures and edge cases. Previously all new workspaces used a generic description and showed “1 project skill(s)”; now Console/Settings/Library pass “Local workspace created from <Surface>.”, and folder rows render “1 project skill” or “N project skills”. (Satisfies TASK-17962 and TASK-17964.)

- Behavior and review notes
  - Activation failures: if `set_active_workspace` raises, Console notifies and skips post-activation sync; Settings surfaces the error inline; the workspace is still created. Library behavior is unchanged and already covered by tests.
  - Modal provenance: `WorkspaceCreateModal(description=...)` stores the caller’s wording; default remains “Created from the workspace setup dialog.” A Google-style docstring documents the param.
  - Input flows: pressing Enter in the name field submits; an unchecked “Switch to this workspace” survives a recompose after adding a folder.
  - TOCTOU: deleting a folder between Add and Create yields an inline failure (`failed_folders`); the workspace is created without bindings.
  - Skills import modal: footer renders “Skipped: …” and “N more not shown”; markup-hostile descriptions render literally; `_import_in_flight` docstring records the accepted no-dismissal posture; off-thread discovery in `_add_folder` is explicitly declined and documented.
  - Types: handler callbacks now accept `WorkspaceCreateResult | None`. No runtime change.
  - No migration required. Callers may optionally pass `description` and should accept a `None` result.

<sup>Written for commit a19dfb0aae8e271c76eb89f03df47f49e7f6d6ae. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/rmusser01/tldw_chatbook/pull/1885?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: tldw_chatbook/Widgets/workspace_create_modal.py [hunk 1]
```diff
@@ -117,9 +117,20 @@ class WorkspaceCreateModal(
     BINDINGS = [("escape", "request_safe_cancel", "Cancel")]
     AUTO_FOCUS = "#workspace-create-name"
 
-    def __init__(self, *, registry_service: LocalWorkspaceRegistryService) -> None:
+    def __init__(
+        self,
+        *,
+        registry_service: LocalWorkspaceRegistryService,
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply tldw_chatbook/Widgets/workspace_create_modal.py:123]: Fixed in a19dfb0aa — `WorkspaceCreateModal.__init__` now carries a Google-style docstring documenting both `registry_service` and the new `description` provenance parameter.

File: Tests/Workspaces/test_console_workspace_create_handler.py [hunk 2]
```diff
@@ -81,6 +84,36 @@ def test_failed_folders_surface_as_warnings(tmp_path):
     assert any("Folder does not exist" in n for n in stub.notifications)
 
 
+class _RaisingActivateRegistry:
+    """Wraps a real registry; ``set_active_workspace`` always raises
+    (TASK-17962: the Console activation-failure seam)."""
+
```
[the remark this answers is not in the document]
Comment (rmusser01) [reply Tests/Workspaces/test_console_workspace_create_handler.py:90]: Declined with rationale: leading-underscore helper classes are this test suite's established convention for module-private fixtures — 1,432 such classes across `Tests/` (`_Stub`, `_FakeStore`, `_HarnessApp`, `_RealActivateStub`…), and this very file already defined `class _Stub:` before this PR. `_RaisingActivateRegistry` follows the file it lives in; renaming just this one would make the file internally inconsistent without making anything clearer. The underscore is signalling test-local privacy, not a naming slip — the PascalCase portion after it is intact.
